MOTIVATION.md

```diff
@@ -0,0 +1,138 @@
+sketched above can be realized.
+
+## Scientific Utility
+
+#### Normalization and batch effect correction.
+
+The DCP combines single-cell data from many different sources. But, batch
+effects in single-cell data are particularly strong. Correct analysis of
+combined data from the HCA requires careful handling of batch and other
+normalization issues. The matrix service could prepare a matrix from multiple
+experiments that is simple and easy to use, but such a matrix may only serve to
+enable false discoveries.
+
+#### Metadata availability.
+
+Much of the engineering work around expression matrices focuses on storing and
+delivering the expression values themselves, but practical analysis of
+expression data often turns much more on metadata. An analyst needs to be able
+to access and operate on annotations of cells, genes, and experiments.
+
+#### Data formats and schemata
+
+Multiple single-cell analysis tools have been developed that depend on
+different representations of expression matrix data and metadata. The
+differences limit interoperability and the matrix service's ability to
+integrate seamlessly many tools in current or future use. A limited set of
+dataframe standards agreed to by the community will be needed.
+
+## Scale
+
+The number of cells with expression data in the HCA DCP will grow by several
+orders of magnitude over time. The matrix service will need to handle this
+growth in data volume.
+
+#### Query responsiveness
+
+The matrix service delivers subsets of HCA expression data in response to
+queries from users. As the HCA and the complexity of supported queries grows,
+the time needed to deliver a matrix could become intolerably long. The matrix
+service will need a scalable, burstable architecture.
+
+#### Distributed computation
+
+Most current tertiary analyses of single-cell expression data can be reasonably
+performed on a single computer, but this will change soon. The matrix service
+should enable analyses distributed computation over expression data.
+
+#### Data representation and localization
+
+As data volume grows, when and how to localize expression data to a client will
+become important considerations. For example, a client should be able to refer
+to subsets or operations on HCA data without unexpectedly localizing a huge
+amount expression data.
+
+## Expressing Queries
+
+#### Separation of query and matrix interfaces
+
+The DCP data storage system has its own evolving query interface that may or
+may not align well with the sorts of query expression matrix consumers want to
+make. The matrix service needs to manage the interaction between between the
+two query interfaces.
+
+#### Query execution ordering
+
+When a client makes a query, the most efficient action may not be to execute
+the query immediately but instead batch or delay operations until  
+
+#### Indicating progress and expected running time
+
+Matrix service clients should be aware of the expected resource consumption and
```

Comment:
What does _should be aware_ mean? Is the service returning a hint the client?

Reply:
Yeah, a hint or progress bar or something like that.